### PR TITLE
Improve rendered functions in Mkdocs

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Improve rendering of functions in mkdocs
- comment out strings that should not be rendered
- ensure functions are grouped appropriately

**This PR improves the rendered functions in mkdocs.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [ ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1388--pyjanitor.netlify.app

File: environment-dev.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
   - mypy
   - natsort
   - numba
-  - numpy
+  - numpy==1.24.4
```
Comment (ericmjl) [inline environment-dev.yml]: Was this change because of NumPy 2.0? If so, I think we should pin to `numpy<2.0` instead.

**Review by ericmjl — APPROVED** [review]
Mostly LGTM! 

Thanks @samukweku :smile:

Pre-approving, with only one minor change depending on the intent of the NumPy pin. Once done, please feel free to merge!

File: environment-dev.yml (hunk 1 above)
Comment (samukweku) [reply environment-dev.yml]: Numba requires less than 1.25. I assume this matters only for development?

**Pull request merged**